対象ウィンドウが閉じられた時にメインウィンドウを初期状態にリセットする

## 概要

対象ウィンドウを選択した状態で、そのウィンドウが閉じられると以下のエラーが発生する：

```
[ERROR] Baketa.Infrastructure.Platform.Windows.WindowsGraphicsCapturer: 🚀 [Issue #193] GPUリサイズキャプチャ例外、通常キャプチャにフォールバック
System.InvalidOperationException: ウィンドウキャプチャが全て失敗: WGC失敗 (Null image) → GDI失敗
```

エラー自体は想定内だが、その後メインウィンドウの状態がウィンドウ選択済みのままになっている。

## 期待する挙動

対象ウィンドウが閉じられた（キャプチャ不能になった）ことを検知し、メインウィンドウをウィンドウ未選択の初期状態（ターゲット、設定、終了のみ表示）にリセットする。

## 現状の挙動

対象ウィンドウが閉じられてもUIは選択済み状態のまま残り、キャプチャエラーが継続する。

## 対応方針

- キャプチャ失敗時にウィンドウハンドルの有効性を確認
- ウィンドウが無効（閉じられた）場合、ウィンドウ選択状態をクリアしUIを初期状態に戻す

--- a/Baketa.Application/DI/Modules/ApplicationModule.cs
+++ b/Baketa.Application/DI/Modules/ApplicationModule.cs
@@ -275,6 +275,9 @@ public sealed class ApplicationModule : ServiceModuleBase
                 // Issue #293: 投機的OCRサービス（オプショナル）
                 var speculativeOcrService = provider.GetService<Baketa.Core.Abstractions.OCR.ISpeculativeOcrService>();
 
+                // [Issue #389] ウィンドウ存在チェック用
+                var windowManagerAdapter = provider.GetService<Baketa.Core.Abstractions.Platform.Windows.Adapters.IWindowManagerAdapter>();
+
                 return new Baketa.Application.Services.Translation.TranslationOrchestrationService(
                     captureService,
                     settingsService,
@@ -287,6 +290,7 @@ public sealed class ApplicationModule : ServiceModuleBase
                     fallbackOrchestrator,
                     licenseManager,
                     speculativeOcrService,
+                    windowManagerAdapter,
                     logger);
             }
             catch (Exception ex)

--- a/Baketa.Application/DI/Modules/CaptureModule.cs
+++ b/Baketa.Application/DI/Modules/CaptureModule.cs
@@ -85,12 +85,15 @@ public sealed class CaptureModule : ServiceModuleBase
                 var logger = provider.GetRequiredService<Microsoft.Extensions.Logging.ILogger<AdaptiveCaptureServiceAdapter>>();
                 var coordinateTransformationService = provider.GetRequiredService<Baketa.Core.Abstractions.Services.ICoordinateTransformationService>();
                 var changeDetectionService = provider.GetService<Baketa.Core.Abstractions.Services.IImageChangeDetectionService>();
+                // [Issue #389] ウィンドウクローズ検知用
+                var eventAggregator = provider.GetService<Baketa.Core.Abstractions.Events.IEventAggregator>();
+                var windowManagerAdapter = provider.GetService<Baketa.Core.Abstractions.Platform.Windows.Adapters.IWindowManagerAdapter>();
 
                 logger.LogDebug("AdaptiveCaptureServiceAdapter インスタンス作成");
                 logger.LogInformation($"🎯 [PHASE_C] EnhancedImageChangeDetectionService統合: {(changeDetectionService != null ? "有効" : "無効")}");
                 logger.LogInformation("🎯 [WIN32_OVERLAY_FIX] CoordinateTransformationService統合: 動的ROIScaleFactor計算対応");
 
-                var adapter = new AdaptiveCaptureServiceAdapter(adaptiveService, logger, coordinateTransformationService, changeDetectionService);
+                var adapter = new AdaptiveCaptureServiceAdapter(adaptiveService, logger, coordinateTransformationService, changeDetectionService, eventAggregator, windowManagerAdapter);
                 logger.LogInformation("AdaptiveCaptureServiceAdapter 登録完了 - Phase C画面変化検知 + WIN32座標変換統合済み");
                 return adapter;
             }

--- a/Baketa.Application/Services/ApplicationInitializer.cs
+++ b/Baketa.Application/Services/ApplicationInitializer.cs
@@ -80,17 +80,17 @@ public class ApplicationInitializer : ILoadingScreenInitializer
             // [Issue #213] Phase 1: ダウンロードとGPUセットアップを並列実行
             // これらは独立した処理のため、並列化することで起動時間を短縮
             _logger.LogInformation("[Issue #213] Phase 1: コンポーネントダウンロードとGPUセットアップを並列実行");
-            ReportProgress("parallel_init", "初期化中（ダウンロード + GPU環境セットアップ）...", isCompleted: false, progress: 0);
+            ReportProgress("parallel_init", "Initializing (Download + GPU environment setup)...", isCompleted: false, progress: 0);
 
             var downloadTask = ExecuteStepAsync(
                 "download_components",
-                "コンポーネントをダウンロードしています...",
+                "Downloading components...",
                 DownloadMissingComponentsAsync,
                 cancellationToken);
 
             var gpuSetupTask = ExecuteStepAsync(
                 "setup_gpu",
-                "GPU環境をチェックしています...",
+                "Checking GPU environment...",
                 SetupGpuEnvironmentAsync,
                 cancellationToken);
 
@@ -100,7 +100,7 @@ public class ApplicationInitializer : ILoadingScreenInitializer
             // [Issue #213] Phase 2: 依存関係解決（シーケンシャル - サービス登録に依存）
             await ExecuteStepAsync(
                 "resolve_dependencies",
-                "依存関係を解決しています...",
+                "Resolving dependencies...",
                 ResolveDependenciesAsync,
                 cancellationToken).ConfigureAwait(false);
 
@@ -114,17 +114,17 @@ public class ApplicationInitializer : ILoadingScreenInitializer
             // [Issue #213] Phase 3: OCRと翻訳エンジンを並列初期化
             // これらは独立しており、並列化することで初期化時間を短縮
             _logger.LogInformation("[Issue #213] Phase 3: OCRと翻訳エンジンを並列初期化");
-            ReportProgress("parallel_engines", "OCR・翻訳エンジンを初期化中...", isCompleted: false, progress: 50);
+            ReportProgress("parallel_engines", "Initializing OCR and translation engines...", isCompleted: false, progress: 50);
 
             var ocrTask = ExecuteStepAsync(
                 "load_ocr",
-                "OCRモデルを読み込んでいます...",
+                "Loading OCR models...",
                 InitializeOcrAsync,
                 cancellationToken);
 
             var translationTask = ExecuteStepAsync(
                 "init_translation",
-                "翻訳エンジンを初期化しています...",
+                "Initializing translation engine...",
                 InitializeTranslationAsync,
                 cancellationToken);
 
@@ -134,7 +134,7 @@ public class ApplicationInitializer : ILoadingScreenInitializer
             // Step 4: UIコンポーネント準備（最後に実行）
             await ExecuteStepAsync(
                 "prepare_ui",
-                "UIコンポーネントを準備しています...",
+                "Preparing UI components...",
                 PrepareUIComponentsAsync,
                 cancellationToken).ConfigureAwait(false);
 
@@ -498,7 +498,7 @@ public class ApplicationInitializer : ILoadingScreenInitializer
         {
             // [Issue #198] 「ダウンロード完了」→「インストール完了」に変更
             // 解凍まで完了してから呼ばれるため、より正確な表現に
-            return $"{e.Component.DisplayName} のインストール完了";
+            return $"{e.Component.DisplayName} installation completed";
         }
 
         if (!string.IsNullOrEmpty(e.ErrorMessage))
@@ -511,7 +511,7 @@ public class ApplicationInitializer : ILoadingScreenInitializer
         var speedMBps = e.SpeedBytesPerSecond / 1024.0 / 1024.0;
 
         var etaStr = e.EstimatedTimeRemaining.HasValue
-            ? $" (残り {e.EstimatedTimeRemaining.Value.TotalSeconds:F0}秒)"
+            ? $" ({e.EstimatedTimeRemaining.Value.TotalSeconds:F0}s remaining)"
             : "";
 
         return $"{e.Component.DisplayName}: {receivedMB:F1}MB / {totalMB:F1}MB ({speedMBps:F1}MB/s){etaStr}";

--- a/Baketa.Application/Services/Capture/AdaptiveCaptureServiceAdapter.cs
+++ b/Baketa.Application/Services/Capture/AdaptiveCaptureServiceAdapter.cs
@@ -1,8 +1,10 @@
 using System.Drawing;
 using System.IO;
 using Baketa.Core.Abstractions.Capture;
+using Baketa.Core.Abstractions.Events;
 using Baketa.Core.Abstractions.Imaging;
 using Baketa.Core.Abstractions.Services;
+using Baketa.Core.Events.EventTypes;
 // 🔥 [PHASE_K-29-G] CaptureOptions統合: Baketa.Core.Models.Capture削除
 using Baketa.Core.Settings;
 using Baketa.Infrastructure.Platform.Adapters;
@@ -19,12 +21,16 @@ public partial class AdaptiveCaptureServiceAdapter(
     IAdaptiveCaptureService adaptiveCaptureService,
     ILogger<AdaptiveCaptureServiceAdapter> logger,
     ICoordinateTransformationService coordinateTransformationService,
-    IImageChangeDetectionService? imageChangeDetectionService = null) : ICaptureService, IDisposable
+    IImageChangeDetectionService? imageChangeDetectionService = null,
+    IEventAggregator? eventAggregator = null,
+    Baketa.Core.Abstractions.Platform.Windows.Adapters.IWindowManagerAdapter? windowManagerAdapter = null) : ICaptureService, IDisposable
 {
     private readonly IAdaptiveCaptureService _adaptiveCaptureService = adaptiveCaptureService ?? throw new ArgumentNullException(nameof(adaptiveCaptureService));
     private readonly ILogger<AdaptiveCaptureServiceAdapter> _logger = logger ?? throw new ArgumentNullException(nameof(logger));
     private readonly ICoordinateTransformationService _coordinateTransformationService = coordinateTransformationService ?? throw new ArgumentNullException(nameof(coordinateTransformationService));
     private readonly IImageChangeDetectionService? _imageChangeDetectionService = imageChangeDetectionService;
+    private readonly IEventAggregator? _eventAggregator = eventAggregator;
+    private readonly Baketa.Core.Abstractions.Platform.Windows.Adapters.IWindowManagerAdapter? _windowManagerAdapter = windowManagerAdapter;
     private ServicesCaptureOptions _currentOptions = new();
     private bool _disposed;
 
@@ -187,12 +193,42 @@ public partial class AdaptiveCaptureServiceAdapter(
         catch (Exception ex)
         {
             _logger.LogError(ex, "適応的ウィンドウキャプチャでエラー");
+
+            // [Issue #389] キャプチャ全戦略失敗時にウィンドウの存在を確認
+            // ウィンドウが閉じられている場合、CaptureFailedEventを発行してUIリセットを促す
+            await PublishCaptureFailedIfWindowClosedAsync(windowHandle, ex).ConfigureAwait(false);
+
             throw;
         }
         finally
         {
             // [Issue #361] セマフォ解放
             _captureSemaphore.Release();
+        }
+    }
+
+    /// <summary>
+    /// [Issue #389] ウィンドウが閉じられている場合にCaptureFailedEventを発行
+    /// </summary>
+    private async Task PublishCaptureFailedIfWindowClosedAsync(IntPtr windowHandle, Exception ex)
+    {
+        if (_eventAggregator == null || _windowManagerAdapter == null) return;
+        if (windowHandle == IntPtr.Zero) return;
+
+        try
+        {
+            var bounds = _windowManagerAdapter.GetWindowBounds(windowHandle);
+            if (bounds == null)
+            {
+                _logger.LogInformation("[Issue #389] キャプチャ全戦略失敗 + ウィンドウが存在しない: Handle=0x{Handle:X} → CaptureFailedEvent発行",
+                    windowHandle.ToInt64());
+                await _eventAggregator.PublishAsync(new CaptureFailedEvent(
+                    Rectangle.Empty, ex, "Target window closed")).ConfigureAwait(false);
+            }
+        }
+        catch (Exception publishEx)
+        {
+            _logger.LogWarning(publishEx, "[Issue #389] CaptureFailedEvent発行中にエラー");
         }
     }
 

--- a/Baketa.Application/Services/Translation/TranslationOrchestrationService.cs
+++ b/Baketa.Application/Services/Translation/TranslationOrchestrationService.cs
@@ -55,7 +55,9 @@ enum TranslationStepResult
     /// <summary>クールダウン中（状態維持、最短待機で再試行）</summary>
     InCooldown,
     /// <summary>エラー発生</summary>
-    Error
+    Error,
+    /// <summary>[Issue #389] 対象ウィンドウが閉じられた（ループ停止）</summary>
+    WindowClosed
 }
 
 /// <summary>
@@ -81,6 +83,9 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
 
     // Issue #293: 投機的OCRサービス（Shot翻訳応答時間短縮）
     private readonly ISpeculativeOcrService? _speculativeOcrService;
+
+    // [Issue #389] ウィンドウ存在チェック用
+    private readonly Baketa.Core.Abstractions.Platform.Windows.Adapters.IWindowManagerAdapter? _windowManagerAdapter;
 
     // 状態管理
     private volatile bool _isAutomaticTranslationActive;
@@ -169,6 +174,7 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
         IFallbackOrchestrator? fallbackOrchestrator = null,
         ILicenseManager? licenseManager = null,
         ISpeculativeOcrService? speculativeOcrService = null,
+        Baketa.Core.Abstractions.Platform.Windows.Adapters.IWindowManagerAdapter? windowManagerAdapter = null,
         ILogger<TranslationOrchestrationService>? logger = null)
     {
         ArgumentNullException.ThrowIfNull(captureService);
@@ -188,6 +194,7 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
         _fallbackOrchestrator = fallbackOrchestrator;
         _licenseManager = licenseManager;
         _speculativeOcrService = speculativeOcrService;
+        _windowManagerAdapter = windowManagerAdapter;
         _logger = logger;
 
         // Issue #293: 投機的OCRサービスが利用可能かログ出力
@@ -976,6 +983,13 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
 
                     var stepResult = await ExecuteAutomaticTranslationStepAsync(cancellationToken).ConfigureAwait(false);
 
+                    // [Issue #389] 対象ウィンドウが閉じられた場合はループ終了
+                    if (stepResult == TranslationStepResult.WindowClosed)
+                    {
+                        _logger?.LogInformation("[Issue #389] 対象ウィンドウが閉じられたため自動翻訳ループを停止します");
+                        break;
+                    }
+
                     // [Issue #394] InCooldown時の残りクールダウン計算（ロック順序を守るため先に取得）
                     TimeSpan remainingCooldown = TimeSpan.Zero;
                     if (stepResult == TranslationStepResult.InCooldown)
@@ -1165,6 +1179,26 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
         var captureStopwatch = System.Diagnostics.Stopwatch.StartNew();
         try
         {
+            // [Issue #389] ウィンドウ存在チェック（キャプチャ前）
+            // キャプチャパイプラインはフォールバック（WGC → GDI）で常にnon-null画像を返すため、
+            // キャプチャ前に明示的にウィンドウの存在を確認する
+            if (_targetWindowHandle.HasValue && _windowManagerAdapter != null)
+            {
+                var bounds = _windowManagerAdapter.GetWindowBounds(_targetWindowHandle.Value);
+                if (bounds == null)
+                {
+                    _logger?.LogInformation("[Issue #389] 対象ウィンドウが閉じられました: Handle=0x{Handle:X}",
+                        _targetWindowHandle.Value.ToInt64());
+
+                    await _eventAggregator.PublishAsync(new CaptureFailedEvent(
+                        System.Drawing.Rectangle.Empty,
+                        new InvalidOperationException($"Target window has been closed: Handle=0x{_targetWindowHandle.Value.ToInt64():X}"),
+                        "Target window closed")).ConfigureAwait(false);
+
+                    return TranslationStepResult.WindowClosed;
+                }
+            }
+
             // 進行状況を通知
             PublishProgress(translationId, TranslationStatus.Capturing, 0.1f, "画面キャプチャ中...");
 
@@ -1184,7 +1218,11 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
                 _logger?.LogDebug($"🔥 [CAPTURE_DEBUG] CaptureWindowAsync呼び出し完了");
                 if (currentImage is null)
                 {
-                    throw new TranslationException("ウィンドウキャプチャに失敗しました");
+                    // [Issue #389] ウィンドウキャプチャ失敗時にCaptureFailedEventを発行
+                    var captureEx = new InvalidOperationException($"Window capture failed: Handle={windowHandle}");
+                    await _eventAggregator.PublishAsync(new CaptureFailedEvent(
+                        System.Drawing.Rectangle.Empty, captureEx, captureEx.Message)).ConfigureAwait(false);
+                    throw new TranslationException("Window capture failed");
                 }
                 _logger?.LogDebug($"📷 ウィンドウキャプチャ完了: {(currentImage is not null ? "成功" : "失敗")}");
             }
@@ -1356,6 +1394,29 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
             currentImage?.Dispose(); // エラー時のリソース破棄
             _logger?.LogError(ex, "自動翻訳ステップでエラーが発生しました");
             PublishProgress(translationId, TranslationStatus.Error, 1.0f, $"エラー: {ex.Message}");
+
+            // [Issue #389] キャプチャ失敗後にウィンドウの存在を確認
+            // キャプチャが全戦略で失敗した場合、ウィンドウが閉じられた可能性が高い
+            if (_targetWindowHandle.HasValue && _windowManagerAdapter != null)
+            {
+                var bounds = _windowManagerAdapter.GetWindowBounds(_targetWindowHandle.Value);
+                if (bounds == null)
+                {
+                    _logger?.LogInformation("[Issue #389] キャプチャ失敗後にウィンドウが存在しないことを確認: Handle=0x{Handle:X}",
+                        _targetWindowHandle.Value.ToInt64());
+                    try
+                    {
+                        await _eventAggregator.PublishAsync(new CaptureFailedEvent(
+                            System.Drawing.Rectangle.Empty, ex, "Target window closed")).ConfigureAwait(false);
+                    }
+                    catch (Exception publishEx)
+                    {
+                        _logger?.LogWarning(publishEx, "[Issue #389] CaptureFailedEvent発行失敗");
+                    }
+                    return TranslationStepResult.WindowClosed;
+                }
+            }
+
             return TranslationStepResult.Error;
         }
 #pragma warning restore CA1031
@@ -1560,6 +1621,24 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
         string translationId,
         CancellationToken cancellationToken)
     {
+        // [Issue #389] ウィンドウ存在チェック（キャプチャ前）
+        if (_targetWindowHandle.HasValue && _targetWindowHandle.Value != IntPtr.Zero && _windowManagerAdapter != null)
+        {
+            var bounds = _windowManagerAdapter.GetWindowBounds(_targetWindowHandle.Value);
+            if (bounds == null)
+            {
+                _logger?.LogInformation("[Issue #389] 対象ウィンドウが閉じられました（Singleshot）: Handle=0x{Handle:X}",
+                    _targetWindowHandle.Value.ToInt64());
+
+                await _eventAggregator.PublishAsync(new CaptureFailedEvent(
+                    System.Drawing.Rectangle.Empty,
+                    new InvalidOperationException($"Target window has been closed: Handle=0x{_targetWindowHandle.Value.ToInt64():X}"),
+                    "Target window closed")).ConfigureAwait(false);
+
+                return (null, null);
+            }
+        }
+
         // キャプチャ処理
         IImage? currentImage;
         if (_targetWindowHandle.HasValue && _targetWindowHandle.Value != IntPtr.Zero)
@@ -1808,6 +1887,19 @@ public sealed class TranslationOrchestrationService : ITranslationOrchestrationS
                 {
                     _logger?.LogDebug($"❌ 座標ベース処理でエラー発生: {coordinateEx.Message}");
                     _logger?.LogDebug($"❌ エラーのスタックトレース: {coordinateEx.StackTrace}");
+
+                    // [Issue #389] ウィンドウが閉じられたことによるエラーの場合はフォールバックせず上位に伝播
+                    if (_targetWindowHandle.HasValue && _windowManagerAdapter != null)
+                    {
+                        var bounds = _windowManagerAdapter.GetWindowBounds(_targetWindowHandle.Value);
+                        if (bounds == null)
+                        {
+                            _logger?.LogInformation("[Issue #389] 座標ベース処理失敗後にウィンドウが存在しないことを確認: Handle=0x{Handle:X}",
+                                _targetWindowHandle.Value.ToInt64());
+                            throw; // 上位のcatchでWindowClosedとして処理される
+                        }
+                    }
+
                     _logger?.LogWarning(coordinateEx, "⚠️ 座標ベース処理でエラーが発生、従来のOCR処理にフォールバック: ID={TranslationId}", translationId);
                     // 座標ベース処理でエラーが発生した場合は従来のOCR処理にフォールバック
                     coordinateBasedTranslationExecuted = false;

--- a/Baketa.Application/Services/UI/WindowManagementService.cs
+++ b/Baketa.Application/Services/UI/WindowManagementService.cs
@@ -214,12 +214,20 @@ public sealed class WindowManagementService : IWindowManagementService, IDisposa
     {
         try
         {
-            // TODO: より詳細なウィンドウ有効性検証ロジックを実装
-            // - ウィンドウが存在するか
-            // - ウィンドウがアクセス可能か
-            // - セキュリティ制限に引っかからないか
+            if (windowInfo.Handle == IntPtr.Zero || string.IsNullOrEmpty(windowInfo.Title))
+                return false;
 
-            return windowInfo.Handle != IntPtr.Zero && !string.IsNullOrEmpty(windowInfo.Title);
+            // [Issue #389] IWindowManagerAdapter.GetWindowBounds() でウィンドウの実在を確認
+            // ウィンドウが閉じられている場合、GetWindowBounds() は null を返す
+            var bounds = _windowManager.GetWindowBounds(windowInfo.Handle);
+            if (bounds == null)
+            {
+                _logger.LogInformation("[Issue #389] ウィンドウが存在しません: '{Title}' (Handle={Handle})",
+                    windowInfo.Title, windowInfo.Handle);
+                return false;
+            }
+
+            return true;
         }
         catch (Exception ex)
         {

--- a/Baketa.Infrastructure.Platform/Windows/Overlay/LayeredOverlayWindow.cs
+++ b/Baketa.Infrastructure.Platform/Windows/Overlay/LayeredOverlayWindow.cs
@@ -737,7 +737,7 @@ public sealed class LayeredOverlayWindow : ILayeredOverlayWindow
     /// <param name="font">使用するフォント</param>
     /// <param name="maxWidth">最大幅（ピクセル）</param>
     /// <returns>分割された行のリスト</returns>
-    private List<string> GetWrappedTextLines(Graphics g, string text, Font font, float maxWidth)
+    private static List<string> GetWrappedTextLines(Graphics g, string text, Font font, float maxWidth)
     {
         var lines = new List<string>();
         var paragraphs = text.Split(new[] { '\n' }, StringSplitOptions.None);
@@ -750,38 +750,89 @@ public sealed class LayeredOverlayWindow : ILayeredOverlayWindow
                 continue;
             }
 
-            var words = paragraph.Split(' ');
-            var wrappedLine = new System.Text.StringBuilder();
-
-            foreach (var word in words)
+            // CJK文字を含む場合は文字単位で折り返す
+            if (ContainsCjkCharacters(paragraph))
             {
-                if (wrappedLine.Length > 0)
-                {
-                    var testLine = wrappedLine.ToString() + " " + word;
-                    if (g.MeasureString(testLine, font).Width > maxWidth)
-                    {
-                        lines.Add(wrappedLine.ToString());
-                        wrappedLine.Clear();
-                        wrappedLine.Append(word);
-                    }
-                    else
-                    {
-                        wrappedLine.Append(" " + word);
-                    }
-                }
-                else
-                {
-                    wrappedLine.Append(word);
-                }
+                WrapTextByCharacter(g, paragraph, font, maxWidth, lines);
             }
-
-            if (wrappedLine.Length > 0)
+            else
             {
-                lines.Add(wrappedLine.ToString());
+                WrapTextByWord(g, paragraph, font, maxWidth, lines);
             }
         }
 
         return lines;
+    }
+
+    /// <summary>
+    /// スペース区切りの単語単位でテキストを折り返す（英語等のラテン文字テキスト用）
+    /// </summary>
+    private static void WrapTextByWord(Graphics g, string paragraph, Font font, float maxWidth, List<string> lines)
+    {
+        var words = paragraph.Split(' ');
+        var wrappedLine = new System.Text.StringBuilder();
+
+        foreach (var word in words)
+        {
+            if (wrappedLine.Length > 0)
+            {
+                var testLine = wrappedLine.ToString() + " " + word;
+                if (g.MeasureString(testLine, font).Width > maxWidth)
+                {
+                    lines.Add(wrappedLine.ToString());
+                    wrappedLine.Clear();
+                    wrappedLine.Append(word);
+                }
+                else
+                {
+                    wrappedLine.Append(" " + word);
+                }
+            }
+            else
+            {
+                wrappedLine.Append(word);
+            }
+        }
+
+        if (wrappedLine.Length > 0)
+        {
+            lines.Add(wrappedLine.ToString());
+        }
+    }
+
+    /// <summary>
+    /// 文字単位でテキストを折り返す（CJK文字を含むテキスト用）
+    /// </summary>
+    private static void WrapTextByCharacter(Graphics g, string paragraph, Font font, float maxWidth, List<string> lines)
+    {
+        var currentLine = new System.Text.StringBuilder();
+        foreach (var ch in paragraph)
+        {
+            var testLine = currentLine.ToString() + ch;
+            if (currentLine.Length > 0 && g.MeasureString(testLine, font).Width > maxWidth)
+            {
+                lines.Add(currentLine.ToString());
+                currentLine.Clear();
+            }
+            currentLine.Append(ch);
+        }
+        if (currentLine.Length > 0)
+            lines.Add(currentLine.ToString());
+    }
+
+    /// <summary>
+    /// テキストにCJK文字（漢字、ひらがな、カタカナ、ハングル）が含まれるかを判定
+    /// </summary>
+    private static bool ContainsCjkCharacters(string text)
+    {
+        foreach (var ch in text)
+        {
+            if (ch >= 0x4E00 && ch <= 0x9FFF) return true;  // CJK統合漢字
+            if (ch >= 0x3040 && ch <= 0x309F) return true;  // ひらがな
+            if (ch >= 0x30A0 && ch <= 0x30FF) return true;  // カタカナ
+            if (ch >= 0xAC00 && ch <= 0xD7AF) return true;  // ハングル
+        }
+        return false;
     }
 
     // ========================================

--- a/Baketa.Infrastructure/Services/BackgroundWarmupService.cs
+++ b/Baketa.Infrastructure/Services/BackgroundWarmupService.cs
@@ -188,21 +188,21 @@ public sealed class BackgroundWarmupService(
 
             var stopwatch = System.Diagnostics.Stopwatch.StartNew();
 
-            ReportProgress(0.0, "ウォームアップ開始", WarmupPhase.Starting);
+            ReportProgress(0.0, "Starting warmup", WarmupPhase.Starting);
 
             // フェーズ1: GPU環境検出（10%）
             await WarmupGpuEnvironmentAsync(cancellationToken).ConfigureAwait(false);
-            ReportProgress(0.1, "GPU環境検出完了", WarmupPhase.GpuDetection);
+            ReportProgress(0.1, "GPU environment detected", WarmupPhase.GpuDetection);
 
             // フェーズ2: OCRエンジン初期化とウォームアップ（50%）
             await WarmupOcrEnginesAsync(cancellationToken).ConfigureAwait(false);
             _isOcrWarmupCompleted = true;
-            ReportProgress(0.6, "OCRウォームアップ完了", WarmupPhase.OcrWarmup);
+            ReportProgress(0.6, "OCR warmup completed", WarmupPhase.OcrWarmup);
 
             // フェーズ3: 翻訳エンジン初期化とウォームアップ（40%）
             await WarmupTranslationEnginesAsync(cancellationToken).ConfigureAwait(false);
             _isTranslationWarmupCompleted = true;
-            ReportProgress(1.0, "全ウォームアップ完了", WarmupPhase.Completed);
+            ReportProgress(1.0, "All warmup completed", WarmupPhase.Completed);
 
             _isWarmupCompleted = true;
             stopwatch.Stop();
@@ -227,7 +227,7 @@ public sealed class BackgroundWarmupService(
             LastError = ex;
 
             _logger.LogError(ex, "ウォームアップ中にエラーが発生しました");
-            ReportProgress(_warmupProgress, $"エラー: {ex.Message}", WarmupPhase.Starting);
+            ReportProgress(_warmupProgress, $"Error: {ex.Message}", WarmupPhase.Starting);
             throw;
         }
     }
@@ -257,7 +257,7 @@ public sealed class BackgroundWarmupService(
     {
         try
         {
-            ReportProgress(0.2, "OCRエンジン初期化中", WarmupPhase.OcrInitialization);
+            ReportProgress(0.2, "Initializing OCR engine", WarmupPhase.OcrInitialization);
 
             // IOcrEngineサービスを取得
             var ocrEngine = _serviceProvider.GetService<IOcrEngine>();
@@ -280,7 +280,7 @@ public sealed class BackgroundWarmupService(
                 }
             }
 
-            ReportProgress(0.4, "OCRエンジンウォームアップ中", WarmupPhase.OcrWarmup);
+            ReportProgress(0.4, "Warming up OCR engine", WarmupPhase.OcrWarmup);
 
             // OCRエンジンウォームアップ
             var warmupSuccess = await ocrEngine.WarmupAsync(cancellationToken).ConfigureAwait(false);
@@ -305,7 +305,7 @@ public sealed class BackgroundWarmupService(
     {
         try
         {
-            ReportProgress(0.7, "翻訳エンジン初期化中", WarmupPhase.TranslationInitialization);
+            ReportProgress(0.7, "Initializing translation engine", WarmupPhase.TranslationInitialization);
 
             // ITranslationEngineサービスを取得（全ての実装をウォームアップ）
             var translationEngines = _serviceProvider.GetServices<ITranslationEngine>().ToList();
@@ -350,7 +350,7 @@ public sealed class BackgroundWarmupService(
                         }
 
                         ReportProgress(currentProgress + progressIncrement,
-                            $"{engine.Name}ウォームアップ完了", WarmupPhase.TranslationWarmup);
+                            $"{engine.Name} warmup completed", WarmupPhase.TranslationWarmup);
                     }
                     catch (Exception ex)
                     {

--- a/Baketa.UI/ViewModels/MainOverlayViewModel.cs
+++ b/Baketa.UI/ViewModels/MainOverlayViewModel.cs
@@ -990,6 +990,9 @@ public class MainOverlayViewModel : ViewModelBase
 
         // 🔥 [Issue #300] リソース監視イベントの購読（システムRAM警告も対応）
         SubscribeToEvent<ResourceMonitoringEvent>(OnResourceMonitoringWarning);
+
+        // [Issue #389] キャプチャ失敗イベントの購読（ウィンドウクローズ検知）
+        SubscribeToEvent<CaptureFailedEvent>(OnCaptureFailed);
     }
 
     private void InitializePropertyChangeHandlers()
@@ -1950,6 +1953,74 @@ public class MainOverlayViewModel : ViewModelBase
                 Logger?.LogInformation("[Issue #300] Memory warning cleared (usage: {Usage:F1}%)", memoryUsage);
             }
         });
+    }
+
+    /// <summary>
+    /// [Issue #389] キャプチャ失敗イベントハンドラー
+    /// 対象ウィンドウがクローズされた場合、翻訳を停止しウィンドウ選択を解除する
+    /// </summary>
+    private async Task OnCaptureFailed(CaptureFailedEvent evt)
+    {
+        Logger?.LogDebug("[Issue #389] CaptureFailedEvent received: {ErrorMessage}", evt.ErrorMessage);
+
+        try
+        {
+            // ウィンドウの有効性を確認
+            var isValid = await _windowManagementService.ValidateSelectedWindowAsync().ConfigureAwait(false);
+            if (isValid)
+            {
+                // ウィンドウはまだ存在する - 一時的なキャプチャエラーなので無視
+                Logger?.LogDebug("[Issue #389] Window is still valid - ignoring capture failure");
+                return;
+            }
+
+            Logger?.LogInformation("[Issue #389] Target window is no longer valid - stopping translation and clearing selection");
+
+            // [Issue #389] セマフォ強制リセット（翻訳中のgRPCタイムアウト対策）
+            Baketa.Application.EventHandlers.Translation.AggregatedChunksReadyEventHandler.ResetSemaphoreForStop();
+
+            // [Issue #389] StopTranslationRequestEventを発行（CTS キャンセルで翻訳ループを停止）
+            try
+            {
+                await PublishEventAsync(new StopTranslationRequestEvent()).ConfigureAwait(false);
+                Logger?.LogDebug("[Issue #389] StopTranslationRequestEvent published");
+            }
+            catch (Exception eventEx)
+            {
+                Logger?.LogWarning(eventEx, "[Issue #389] StopTranslationRequestEvent publish failed");
+            }
+
+            // 翻訳オーバーレイを非表示
+            try
+            {
+                await _overlayManager.HideAllAsync().ConfigureAwait(false);
+            }
+            catch (Exception overlayEx)
+            {
+                Logger?.LogWarning(overlayEx, "[Issue #389] Overlay hide failed");
+            }
+
+            // ウィンドウ選択を解除
+            await _windowManagementService.ClearWindowSelectionAsync().ConfigureAwait(false);
+
+            // UI状態を完全にリセット（初期状態に戻す）
+            await Avalonia.Threading.Dispatcher.UIThread.InvokeAsync(() =>
+            {
+                IsTranslationActive = false;
+                IsLoading = false;
+                IsTranslationResultVisible = false;
+                IsSingleshotOverlayVisible = false;
+                SelectedWindow = null;
+                IsWindowSelected = false;
+                CurrentStatus = TranslationStatus.Idle;
+                Logger?.LogDebug("[Issue #389] UI state fully reset to initial state");
+            });
+
+        }
+        catch (Exception ex)
+        {
+            Logger?.LogError(ex, "[Issue #389] Error handling capture failure");
+        }
     }
 
     /// <summary>

--- a/Baketa.UI/Views/ToastNotificationWindow.axaml.cs
+++ b/Baketa.UI/Views/ToastNotificationWindow.axaml.cs
@@ -89,13 +89,15 @@ public partial class ToastNotificationWindow : Window
             if (screen == null) return;
 
             var workingArea = screen.WorkingArea;
+            var scaling = screen.Scaling;
             const int margin = 16;
 
-            // 実際のウィンドウサイズを取得（レイアウト完了後なので正確）
-            var windowHeight = (int)Bounds.Height;
+            // 実際のウィンドウサイズを取得し、DPIスケーリングで物理ピクセルに変換
+            // Bounds.Height は論理ピクセル、WorkingArea は物理ピクセルのため変換が必要
+            var windowHeight = (int)(Bounds.Height * scaling);
             if (windowHeight <= 0)
             {
-                windowHeight = 100; // フォールバック値
+                windowHeight = (int)(100 * scaling); // フォールバック値
             }
 
             // [Issue #343] WorkingArea.X/Y を考慮（マルチモニター対応）

--- a/tests/Baketa.Application.Tests/Services/Translation/TranslationOrchestrationServiceTests.cs
+++ b/tests/Baketa.Application.Tests/Services/Translation/TranslationOrchestrationServiceTests.cs
@@ -90,6 +90,7 @@ public class TranslationOrchestrationServiceTests : IDisposable
             null, // fallbackOrchestrator（Issue #290: テスト用にnull）
             null, // licenseManager（Issue #290: テスト用にnull）
             null, // speculativeOcrService（Issue #293: テスト用にnull）
+            null, // windowManagerAdapter（Issue #389: テスト用にnull）
             _loggerMock.Object);
     }
 

